src/lasp_plumtree_broadcast_distribution_backend.erl

```diff
@@ -325,7 +347,19 @@ bind_to(Id, TheirId) ->
 %%
 -spec read(id(), threshold()) -> {ok, var()} | error().
 read(Id, Threshold) ->
-    gen_server:call(?MODULE, {read, Id, Threshold}, infinity).
+    case gen_server:call(?MODULE, {read, Id, Threshold}, infinity) of
+        {error, not_found} ->
+            case Id of
+                {StorageId, TypeId} ->
+                    declare(StorageId, TypeId),
+                    gen_server:call(?MODULE, {read, Id, Threshold}, infinity);
```

Comment:
Similar comments above about the two step process -- can we just make the update implicitly call the declare to make this more streamlined?